### PR TITLE
Enables middleware to work with runtimeVersion

### PR DESCRIPTION
**Pull request by andreban**

### CODE REVIEW

File: ssr-express/lib/AmpSsrMiddleware.js [hunk 1]
```diff
@@ -97,10 +107,23 @@ class AmpSsrMiddleware {
         // This is a request for the canonical URL. Generate the AMP equivalent
         // in order to add it to the link rel tag.
         const linkRelAmpHtmlUrl = urlMapping.toAmpUrl(req.url);
-        ssr.transformHtml(body, {ampUrl: linkRelAmpHtmlUrl})
+
+        runtimeVersion()
+          .then(version => version)
```
Comment (sebastianbenz) [inline ssr-express/lib/AmpSsrMiddleware.js]: Is this needed?

File: ssr-express/lib/AmpSsrMiddleware.js [hunk 2]
```diff
@@ -97,10 +107,23 @@ class AmpSsrMiddleware {
...
+          .then(version => ssr.transformHtml(body, {
+            ampUrl: linkRelAmpHtmlUrl,
+            ampRuntimeVersion: version
+          }))
           .then(transformedBody => {
-            res.write(Buffer.from(transformedBody, 'utf-8'));
-            res.end();
+            res.status(200).send(transformedBody);
+          })
+          .catch(err => {
```
Comment (sebastianbenz) [inline ssr-express/lib/AmpSsrMiddleware.js]: Do multiple catch handlers work in a promise chain?

File: ssr-express/lib/AmpSsrMiddleware.js (hunk 1 above)
Comment (andreban) [reply ssr-express/lib/AmpSsrMiddleware.js]: Actually, no - removed.

File: ssr-express/lib/AmpSsrMiddleware.js (hunk 2 above)
Comment (andreban) [reply ssr-express/lib/AmpSsrMiddleware.js]: They do!

Comment (sebastianbenz) [reply ssr-express/lib/AmpSsrMiddleware.js]: Will the promise chain stop after the first handler?

Comment (andreban) [reply ssr-express/lib/AmpSsrMiddleware.js]: The first `catch` handler is returning null, which will be passed to the `then` handler after it. So, the version will be null if retrieving the runtimeVersion fails.

If any error happens from then on, the second `catch` handler will be triggered.

Comment (andreban) [reply ssr-express/lib/AmpSsrMiddleware.js]: The last test on the Spec is testing this behaviour.

**Comment by sebastianbenz** [comment]
Are you planning to update the sample in this PR as well?

**Comment by andreban** [comment]
Sorry, I missed the first comment. Do you think it's worth having a demo with the simple usage and another with all the custom options?

**Comment by sebastianbenz** [comment]
Not sure If we need all custom options, but we should at least have one
with versioned runtime.

**Comment by andreban** [comment]
Cool. I'll move the existing demo to demos/simple and create a demos/runtime-version. But I'd rather do it on another PR.

**Pull request merged**